docs/docs/remark-ast.md

```diff
@@ -0,0 +1,270 @@
+
+## What will be learned in this tutorial
+
+- How to learn how to understand the remark AST
+- How to integrate plugins with `gatsby-transformer-remark`
+- How to manipulate the remark AST to add additional functionality.
+
+## Understanding the Syntax Tree
+
+To get an understanding at what is available in the Markdown AST, take a look at the markdown ast spec that is used in remark and other unist projects: [syntax-tree/mdast](https://github.com/syntax-tree/mdast).
+
+Starting out with a markdown file as below
+
+```markdown
+# Hello World!
+
+This is a [Real page](https://google.com)
+```
+
+the generated `markdownAST` would be as appears:
+
+```JSON
+{
+  "type": "root",
+  "children": [
+    {
+      "type": "heading",
+      "depth": 1,
+      "children": [
+        {
+          "type": "text",
+          "value": "Hello World!",
+          "position": {
+            "start": { "line": 1, "column": 3, "offset": 2 },
+            "end": { "line": 1, "column": 15, "offset": 14 },
+            "indent": []
+          }
+        }
+      ],
+      "position": {
+        "start": { "line": 1, "column": 1, "offset": 0 },
+        "end": { "line": 1, "column": 15, "offset": 14 },
+        "indent": []
+      }
+    },
+    {
+      "type": "paragraph",
+      "children": [
+        {
+          "type": "text",
+          "value": "This is a ",
+          "position": {
+            "start": { "line": 3, "column": 1, "offset": 16 },
+            "end": { "line": 3, "column": 11, "offset": 26 },
+            "indent": []
+          }
+        },
+        {
+          "type": "link",
+          "title": null,
+          "url": "https://google.com",
+          "children": [
+            {
+              "type": "text",
+              "value": "Real page",
+              "position": {
+                "start": { "line": 3, "column": 12, "offset": 27 },
+                "end": { "line": 3, "column": 21, "offset": 36 },
+                "indent": []
+              }
+            }
+          ],
+          "position": {
+            "start": { "line": 3, "column": 11, "offset": 26 },
+            "end": { "line": 3, "column": 42, "offset": 57 },
+            "indent": []
+          }
+        }
+      ],
+      "position": {
+        "start": { "line": 3, "column": 1, "offset": 16 },
+        "end": { "line": 3, "column": 42, "offset": 57 },
+        "indent": []
+      }
+    }
+  ],
+  "position": {
+    "start": { "line": 1, "column": 1, "offset": 0 },
+    "end": { "line": 4, "column": 1, "offset": 58 }
+  }
+}
+```
+
+This is the syntax tree that will be parsed by remark into HTML by `gatsby-transformer-remark`.
```

Comment:
How about adding a link to [AST explorer](https://astexplorer.net/#/gist/d9029a2e8827265fbb9b190083b59d4d/3384f3ce6a3084e50043d0c8ce34628ed7477603) here?

Reply:
Removed the large AST block and linked to the AST explorer as provided